packages/serverpod/lib/src/database/database_query.dart

```diff
@@ -627,8 +925,14 @@ String _joinStatementFromJoinContexts(
     LinkedHashMap<String, _JoinContext> joinContexts) {
   List<String> joinStatements = [];
   for (var joinContext in joinContexts.values) {
+    var joinStatement = joinContext.joinStatement;
+    if (joinStatement != null) {
+      joinStatements.add(joinStatement);
+      continue;
+    }
+
     var tableRelation = joinContext.tableRelation;
-    var joinStatement = 'LEFT JOIN';
+    joinStatement = 'LEFT JOIN';
     if (!joinContext.subQuery) {
       joinStatement += ' "${tableRelation.lastForeignTableName}" AS';
     }
```

Comment:
This will be part of the refactoring of removing `_JoinContext` completely. This code here shouldn't be necessary after that fix. Is it OK to hold off and do it in a separate PR?

A couple of lines above we actually opt out of this process all together if the join context contains a `joinStatement`. That couple the logic to the specific cases instead of having an add hoc shared logic for it.

Reply:
Alright! If we are gonna remove it with the other filters then it is fine for now.